mewarpx/tests/test_field_diag.py

```diff
@@ -0,0 +1,131 @@
+"""
+Test that verifies data is plotted correctly from the
+Monte-Carlo Collision script based on case 1 from
+Turner et al. (2013) - https://doi.org/10.1063/1.4775084
+"""
+from matplotlib.pyplot import plot
+import pytest
+import os.path
+
+from mewarpx import util as mwxutil
+
+from pywarpx import picmi
+
+import numpy as np
+
+constants = picmi.constants
+
+@pytest.mark.parametrize("plot_on_diag_steps, post_processing",
+[
+    (True, False),
+    (False, True)
+])
+def test_field_diag(plot_on_diag_steps, post_processing):
+    dim = 2
+    use_rz = False
+
+    mwxutil.init_libwarpx(ndim=dim, rz=use_rz)
+
+    from mewarpx.mwxrun import mwxrun
+    from mewarpx.diags_store import diag_base
+    from mewarpx.setups_store import diode_setup
+    from mewarpx import testing_util
+
+    test_name = "diag_test_"
+    if plot_on_diag_steps:
+        test_name += "with_"
+    else:
+        test_name += "without_"
+    test_name += "diag_plotting_"
+    if post_processing:
+        test_name += "with_"
+    else:
+        test_name += "without_"
+    test_name += "post_processing"
```

Comment:
Tons of ways to do this, maybe to make it a little more concise, do
```suggestion
    plot_diag_str = "_with_diag_plotting" if plot_on_diag_steps else ""
    post_proc_str = "_with_post_processing" if post_processing else ""
    test_name = "field_diag_test" + plot_diag_str + post_proc_str
```